pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -453,8 +455,13 @@ private BrokerResponseNative handleRequest(long requestId, String query, JsonNod
 
       // Send empty response since we don't need to evaluate either offline or realtime request.
       BrokerResponseNative brokerResponse = BrokerResponseNative.empty();
+      // Extract source info from incoming request
+      Collection<String> remoteIps = null;
```

Comment:
nit: code is repeated. Also, we do this processing even if the log is not printed (because of rateLimiter). May be just pass in requesterIdentity to logBrokerResponse() and perform this computation only if we are going to log the metrics?

Reply:
requesterIdentity has a lot of params can mean different things based on which class is being used. Actually does it make sense to skip logging in this particular case ?

Reply:
My suggestion is to pass requestIdentity to logBrokerResponse() instead of passing clientIps. 

`logBrokerResponse(long requestId, ..., @Nullable RequesterIdentity requesterIdentity) `

In logBrokerResponse, we can extract clientIps only if the property is set and rateLimiter decides to log the query.

Reply:
I had this implementation originally, modified based on feedback : https://github.com/apache/pinot/pull/8907#discussion_r899427901